tests/network/vmi_istio.go

```diff
@@ -261,25 +267,54 @@ var _ = SIGDescribe("[Serial] Istio", func() {
 				serverVMI = libvmi.NewCirros(
 					libvmi.WithNetwork(v1.DefaultPodNetwork()),
 					libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding([]v1.Port{}...)),
+					libvmi.WithLabel("version", "v1"),
+					libvmi.WithLabel("app", vmiServerAppSelector),
 				)
 				serverVMI, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(serverVMI)
 				Expect(err).ToNot(HaveOccurred())
-				Expect(console.LoginToCirros(serverVMI)).To(Succeed())
 
+				serverVMIService := newService(
+					"vmi-server",
+					map[string]string{"app": vmiServerAppSelector},
+					[]corev1.ServicePort{{Port: vmiServerTestPort}},
+				)
+				_, err = virtClient.CoreV1().Services(util.NamespaceTestDefault).Create(context.Background(), serverVMIService, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(console.LoginToCirros(serverVMI)).To(Succeed())
 				By("Starting HTTP Server")
-				tests.StartHTTPServer(serverVMI, testPort)
+				tests.StartHTTPServer(serverVMI, vmiServerTestPort)
+
+				By("Creating Istio VirtualService")
+				virtualServicesRes := schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "virtualservices"}
+				virtualService := generateVirtualService()
+				_, err = virtClient.DynamicClient().Resource(virtualServicesRes).Namespace(util.NamespaceTestDefault).Create(context.TODO(), virtualService, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
 
-				By("Getting back the Server VMI IP")
-				serverVMI, err = virtClient.VirtualMachineInstance(serverVMI.Namespace).Get(serverVMI.Name, &metav1.GetOptions{})
+				By("Creating Istio DestinationRule")
+				destinationRulesRes := schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "destinationrules"}
+				destinationRule := generateDestinationRule()
+				_, err = virtClient.DynamicClient().Resource(destinationRulesRes).Namespace(util.NamespaceTestDefault).Create(context.TODO(), destinationRule, metav1.CreateOptions{})
 				Expect(err).ShouldNot(HaveOccurred())
-				serverVMIAddress = serverVMI.Status.Interfaces[0].IP
+
+				By("Creating Istio Gateway")
+				gatewaysRes := schema.GroupVersionResource{Group: "networking.istio.io", Version: "v1beta1", Resource: "gateways"}
+				gateway := generateGateway()
+				_, err = virtClient.DynamicClient().Resource(gatewaysRes).Namespace(util.NamespaceTestDefault).Create(context.TODO(), gateway, metav1.CreateOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+
+				By("Getting back Istio ingressgateway IP")
```

Comment:
nit: why "getting back" and not just "getting" ?

Reply:
I don't know either, Done